core/graphelier-service/db/mongo.go

```diff
@@ -29,20 +28,64 @@ func NewConnection() (*Connector, error) {
 	if err = client.Ping(context.TODO(), nil); err != nil {
 		log.Fatal(err)
 	}
-	fmt.Println("Connected to MongoDB :)")
+	log.Println("Connected to MongoDB :)")
 
 	return &Connector{client}, nil
 }
 
-// FindOrderbook : Finds the Orderbook of an instrument based on the timestamp requested in the db
-func (c *Connector) FindOrderbook(instrument string, timestamp float64) (result *models.Orderbook) {
+// GetOrderbook : Finds the Orderbook of an instrument based on the timestamp requested in the db
+func (c *Connector) GetOrderbook(instrument string, timestamp uint64) (result *models.Orderbook) {
 	collection := c.Database("graphelier-db").Collection("orderbooks")
-	filter := bson.D{{Key: "instrument", Value: instrument}, {Key: "timestamp", Value: timestamp}}
+	filter := bson.D{
+		{Key: "instrument", Value: instrument},
+		{Key: "timestamp", Value: bson.D{
+			{Key: "$lte", Value: timestamp},
+		}},
+	}
+	option := options.FindOne()
+	option.SetSort(bson.D{{Key: "timestamp", Value: -1}})
 
-	err := collection.FindOne(context.TODO(), filter).Decode(&result)
+	err := collection.FindOne(context.TODO(), filter, option).Decode(&result)
 	if err != nil {
 		log.Fatal(err)
 	}
 
 	return result
 }
+
+// GetMessages : Finds the Message of an instrument based on the timestamp requested
+func (c *Connector) GetMessages(instrument string, timestamp uint64, latestFullSnapshot uint64) (results []*models.Message) {
```

Comment:
this should return a tuple in the form of ([]*models.Message, error) to allow for error handling

Reply:
whenever there is an error along the execution you have to return (nil, error object) to stop the function from crashing in the cases where there is an error

Reply:
if the execution is successful return ([]*models.Message, nil)

Reply:
the callers of this function can then handle the errors if there are any